test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -1226,6 +1226,62 @@ def test_circuit_factors(self):
         """Test number of separable factors in circuit."""
         self.assertEqual(self.dag.num_tensor_factors(), 2)
 
+    def test_separable_circuits(self):
+        """Test separating disconnected sets of qubits in a circuit."""
+        # Empty case
+        dag = DAGCircuit()
+        self.assertEqual([], dag.separable_circuits())
+
+        # 3 disconnected qubits
+        qreg = QuantumRegister(3, "qr")
+        creg = ClassicalRegister(2, "cr")
+        dag.add_qreg(qreg)
+        dag.add_creg(creg)
+        dag.apply_operation_back(HGate(), [qreg[0]], [])
+        dag.apply_operation_back(HGate(), [qreg[1]], [])
+        dag.apply_operation_back(HGate(), [qreg[2]], [])
+        dag.apply_operation_back(HGate(), [qreg[2]], [])
+        self.assertEqual(3, len(dag.separable_circuits(remove_idle_wires=True)))
+
+        # 2 sets of disconnected qubits
+        dag.apply_operation_back(CXGate(), [qreg[1], qreg[2]], [])
+        self.assertEqual(2, len(dag.separable_circuits(remove_idle_wires=True)))
+
+        # One connected component
+        dag.apply_operation_back(CXGate(), [qreg[0], qreg[1]], [])
+        self.assertEqual(1, len(dag.separable_circuits(remove_idle_wires=True)))
```

Comment:
I'm having problems when I turn `remove_idle_wires` to `True`, because `dag.remove_qubits` is reformatting the names of the qubits, which is causing my comparisons to be off. The new qubit name format after removing qubits is `q_N`, and the old format is `q_[N]`.  I'm not sure how to recreate that format, as qubit names seems to always be appended with `[N]`

Reply:
I'm not sure I understand: if I build a couple of DAGs manually, the equality comparisons seem to work the way I expect:

```python
from qiskit.circuit import QuantumRegister
from qiskit.dagcircuit import DAGCircuit

qr = QuantumRegister(3)
a = DAGCircuit()
a.add_qreg(qr)
a.remove_qubits(qr[2])

b = DAGCircuit()
b.add_qubits([qr[0], qr[1]])

assert a == b
```

Reply:
Thank you, I was doing it in a silly way.